jsonrpc/http.go

```diff
@@ -55,7 +55,15 @@ func (h *HTTP) Run(ctx context.Context) error {
 
 // ServeHTTP processes an incoming HTTP request
 func (h *HTTP) ServeHTTP(writer http.ResponseWriter, req *http.Request) {
-	if req.Method != "POST" {
+	if req.Method == "GET" {
+		status := http.StatusNotFound
+		if req.URL.Path == "/" {
+			status = http.StatusOK
+		}
+		writer.WriteHeader(status)
+		req.Close = true
```

Comment:
[Looks like](https://pkg.go.dev/net/http#Request:~:text=//%20For%20server%20requests%2C%20the%20HTTP%20server%20handles%20this%20automatically) in server handlers we don't need to set it explicitly